CRM/Core/Form.php

```diff
@@ -1688,7 +1688,8 @@ public function addSelect($name, $props = [], $required = FALSE) {
           break;
         }
       }
-      $label = $props['label'] ?? $fieldSpec['title'];
+      $fieldSpecLabel = $fieldSpec['html']['label'] ?? CRM_Utils_Array::value('title', $fieldSpec);
+      $label = CRM_Utils_Array::value('label', $props, $fieldSpecLabel);
```

Comment:
This can be written more simply and without the ugly `CRM_Utils_Array::value()` function:
```suggestion
       $label = $props['label'] ?? $fieldSpec['html']['label'] ?? $fieldSpec['title'];
```

Reply:
Yup, will use this in the place below where I copy-pasted this from as well.